dependencies.yaml

```diff
@@ -333,37 +336,51 @@ dependencies:
           # Allow runtime version to float up to minor version
           # Disallow pyarrow 14.0.0 due to a CVE
           - pyarrow>=14.0.1,<15.0.0a0
-  cudatoolkit:
+  cuda_version:
     specific:
       - output_types: conda
         matrices:
           - matrix:
-              cuda: "12.*"
+              cuda: "11.2"
```

Comment:
Can we drop 11.2 from this file? We still support 11.2 via CEC, but we don't need to be able to build envs with it any more do we?

Reply:
Perhaps not really in scope for this PR. Something we can come back to later.

Reply:
If we are supporting it, wouldn't we want a way to construct an environment with this CUDA version for testing, debugging, etc.?

Reply:
11.2 has been dropped from our CI matrix. I think it’s OK to remove this but I agree it’s probably better to keep the scope narrow here. At some point we will drop CUDA 11 entirely and clean up a lot of stuff.

Reply:
> Something we can come back to later.

I think it should come later, yeah.